### PR TITLE
[18.0][MIG] base_technical_features: Migration to 18.0

### PR DESCRIPTION
**Pull request by Kimkhoi3010**
I found only 1 commit to port but it was ignored, I added it to blacklist:: [[17.0][FW] base_technical_features: port from 14.0](https://github.com/Kimkhoi3010/server-ux/pull/3)

# Reason for change:
The original logic in _postprocess_access_rights removes base.group_no_one for non-admin users, which restricts their access to certain UI components. Extending _get_view ensures that technical users (part of base_technical_features.group_technical_features) still have access to these components without needing full admin privileges.

### CODE REVIEW

File: base_technical_features/models/base.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# © 2016 Opener B.V. (<https://opener.am>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import models
+
+
+class Base(models.AbstractModel):
+    _inherit = "base"
+
+    def user_has_groups(self, groups):
```
Comment (nilshamerlinck) [inline base_technical_features/models/base.py]: `user_has_groups` has been removed in 18: see `https://github.com/odoo/odoo/pull/151597`

File: base_technical_features/models/ir_ui_view.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+from odoo import api, models
+from odoo.http import request
+
+
+class View(models.Model):
+    _inherit = "ir.ui.view"
+
+    @api.model
+    def _postprocess_access_rights(self, tree):
```
Comment (nilshamerlinck) [inline base_technical_features/models/ir_ui_view.py]: an explanation of the new approach in the PR description would be welcome to help reviewer

File: base_technical_features/models/base.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
 class Base(models.AbstractModel):
     _inherit = "base"
 
-    def user_has_groups(self, groups):
+    def has_groups(self, groups):
```
Comment (nguyenminhchien) [inline base_technical_features/models/base.py]: this method should be in res.users and consider to override `def has_group()` or `def _has_group()`

Comment (nguyenminhchien) [reply base_technical_features/models/base.py]: this commit made me confused, these changes should be included in a migration commit

File: base_technical_features/models/ir_ui_view.py (hunk 2 above)
Comment (nguyenminhchien) [inline base_technical_features/models/ir_ui_view.py]: Overriding this method conflicts with the module `web_studio`

**Review by nguyenminhchien — CHANGES_REQUESTED** [review]

File: base_technical_features/models/ir_ui_view.py (hunk 2 above)
Comment (nguyenminhchien) [reply base_technical_features/models/ir_ui_view.py]: @Kimkhoi3010 
If you cannot find a better way. try this one 
```
    @api.model
    def _postprocess_access_rights(self, tree):
        arch = etree.tostring(tree, encoding="unicode").replace('\t', '').replace('base.group_no_one', 'base_technical_features.group_technical_features')
        tree = etree.fromstring(arch)
        return super()._postprocess_access_rights(tree)
        
```

File: base_technical_features/models/ir_ui_view.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        access rights and are not restricted by the usual access conditions
+        related to the 'group_no_one' group.
+        """
+        arch = (
+            etree.tostring(tree, encoding="unicode")
+            .replace("\t", "")
+            .replace(
+                str(self.env.ref("base.group_no_one").id),
+                str(
+                    self.env.ref("base_technical_features.group_technical_features").id
```
Comment (nguyenminhchien) [inline base_technical_features/models/ir_ui_view.py]: @Kimkhoi3010 replace the string `base.group_no_one` by the string `base_technical_features.group_technical_features`.

NOT replace its database `id`

File: base_technical_features/readme/CREDITS.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+The migration of this module from 17.0 to 18.0 was financially supported by:
+
+- Camptocamp.
```
Comment (nguyenminhchien) [inline base_technical_features/readme/CREDITS.md:3]: Just keep it as before: `The migration of this module from 17.0 to 18.0 was financially supported by Camptocamp.`

File: base_technical_features/tests/test_base_technical_features.py [hunk 6]
```diff
@@ -32,8 +32,11 @@ def get_partner_field():
             return xml.xpath('//div/group/field[@name="partner_id"]')
 
         self.env.user.write({"technical_features": False})
+        print("\n====", self.env.user.groups_id)
         self.assertEqual(len(get_partner_field()), 0)
         self.env.user.write({"technical_features": True})
+        print("\n====", self.env.user.groups_id)
```
Comment (nguyenminhchien) [inline base_technical_features/tests/test_base_technical_features.py]: remove 2 print commands

**Review by nguyenminhchien — CHANGES_REQUESTED** [review]

**Comment by nguyenminhchien** [comment]
seems that you cannot port this commit in https://github.com/OCA/server-ux/pull/411
so, add it to port blacklist by creating a new commit, include that new commit into this PR

**Review by nguyenminhchien — APPROVED** [review]

File: base_technical_features/models/ir_ui_view.py (hunk 2 above)
Comment (nilshamerlinck) [reply base_technical_features/models/ir_ui_view.py]: still no explanation

**Comment by Kimkhoi3010** [comment]
I created a new PR [OCA#953](https://github.com/OCA/server-ux/pull/953) to OCA/server-ux.
(This PR was closed auto after rebased)

**Pull request closed without merging**